src/api.rs

```diff
@@ -11,13 +12,15 @@ use serde::{Deserialize, Serialize};
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
 pub struct Request {
+    pub request: String,
```

Comment:
We want an `Enum` here, which helps us to distinguish between different request types on the, well, *type* level.

```rust
pub enum Request {
    AccountList,
    Code { account: String },
}
```

There are multiple options for serialization/deserialization https://serde.rs/enum-representations.html

For example, we can use `#[serde(tag = "type")]` which will result in representations

```json
{"type": "Request"}
```
and
```json
{"type": "Code", "account": "example.org"}
```